### ISSUE TITLE
[ace:x86-windows] build failure

### ISSUE THREAD

**Issue by tan-wei** [microsoft/vcpkg#16551]
**Host Environment**
- OS: Windows 10
- Compiler: MSVC 16.8.5

**To Reproduce**
`./vcpkg install ace[ssl,tao,wchar,xml,zlib]`

**Failure logs** 
```
...
-- Installing: C:/vcpkg/packages/ace_x86-windows/include/tao/Compression/zlib/ZlibCompressor.cpp
-- Installing: C:/vcpkg/packages/ace_x86-windows/include/tao/Compression/zlib/ZlibCompressor.h
-- Installing: C:/vcpkg/packages/ace_x86-windows/include/tao/Compression/zlib/ZlibCompressor_Factory.cpp
-- Installing: C:/vcpkg/packages/ace_x86-windows/include/tao/Compression/zlib/ZlibCompressor_Factory.h
-- Installing: C:/vcpkg/packages/ace_x86-windows/include/tao/Compression/zlib/ZlibCompressor_export.h
-- Performing post-build validation
Mismatching number of debug and release binaries. Found 167 for debug but 158 for release.
Debug binaries

    C:/vcpkg/packages/ace_x86-windows/debug/lib/ACEd.lib
    C:/vcpkg/packages/ace_x86-windows/debug/lib/ACEXMLd.lib
    C:/vcpkg/packages/ace_x86-windows/debug/lib/ACEXML_Parserd.lib
    C:/vcpkg/packages/ace_x86-windows/debug/lib/ACE_Compressiond.lib
    C:/vcpkg/packages/ace_x86-windows/debug/lib/ACE_ETCLd.lib
...
Found 1 error(s). Please correct the portfile:
    C:\vcpkg\ports\ace\portfile.cmake
-- Performing post-build validation done
Error: Building package ace:x86-windows failed with: POST_BUILD_CHECKS_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: ace:x86-windows
  Vcpkg version: 2021-01-13-d67989bce1043b98092ac45996a8230a059a2d7e

Additionally, attach any relevant sections from the log files above.
```

**Additional context**
 - Package: ace:x86-windows
 - Vcpkg version: 2021-01-13-d67989bce1043b98092ac45996a8230a059a2d7e

**Comment by NancyLi1013** [comment]
Hi @tan-wei 

Thanks for posting this issue.

The following libs are missing in `/lib `folder.

```
F:/vcpkg/packages/ace_x86-windows/debug/lib/tao_cosconcurrency.lib
F:/vcpkg/packages/ace_x86-windows/debug/lib/tao_cosevent.lib
F:/vcpkg/packages/ace_x86-windows/debug/lib/tao_coslifecycle.lib
F:/vcpkg/packages/ace_x86-windows/debug/lib/tao_cosnaming.lib
F:/vcpkg/packages/ace_x86-windows/debug/lib/tao_cosnotification.lib
F:/vcpkg/packages/ace_x86-windows/debug/lib/tao_costrading.lib
F:/vcpkg/packages/ace_x86-windows/debug/lib/tao_imr_activator.lib
F:/vcpkg/packages/ace_x86-windows/debug/lib/tao_imr_locator.lib
F:/vcpkg/packages/ace_x86-windows/debug/lib/tao_rtevent.lib
```

**Comment by tan-wei** [comment]
Waiting for the PR is merged. Thanks.

**Comment by jwillemsen** [comment]
This is very likely because of lack for zlib on windows as part of MPC. Did make a PR for MPC with a similar approach as we use for xercesc/openssl, extend the github actions we use to also test this, see https://github.com/DOCGroup/MPC/pull/128. When that works I can merge it and it will be part of a future ACE version

**Comment by jwillemsen** [comment]
It is not zlib, it looks in a debug build we get some additional libraries which are not there in a release build. This looks to be related to the executables we link. 

At that moment a /IMPLIB commandline option is used but the `d` postfix is not added there. This is something that needs structural work for which I lack free time, see https://www.dre.vanderbilt.edu/~schmidt/commercial-support.html for an overview of the commercial companies including the company I work for as alternative.

**Comment by jwillemsen** [comment]
In a release build we get tao_cosnaming.lib twice, once for the exe, once for the shared library, they overwrite each other when copied

In debug mode we get `tao_cosnaming.lib` for the exe but `tao_cosnamingd.lib` for the shared library so there is a difference amount of shared libraries

**Comment by jwillemsen** [comment]
This should be fixed in the upcoming ACE 7.0.5 release

**Comment by tan-wei** [comment]
@jwillemsen Thanks. Great news!

**Comment by jwillemsen** [comment]
You should thank @mitza-oci, he sorted out the issue!

**Comment by tan-wei** [comment]
Thanks @mitza-oci. :-P

**Comment by JonLiu1993** [comment]
@tan-wei ,Thanks for posting this issue, This issue has been fixed by #21322, please 'git pull' your vcpkg and try again, I am closing this issue for now
